Rebild Main structure of programm:

8Invader8 opened the pull request:
-Make Menu;
-Make FirstOperation(hide inner operetions);
-Make Class Constants;
and another small changes.

Commit:
Rebild Main structure of programm:
-Make Menu;
-Make FirstOperation(hide inner operetions);
-Make Class Constants;
and another small changes.

Volodymyr95 commented on src/Constants.java:
```diff
@@ -0,0 +1,11 @@
+public class Constants {
+    public static final Scanner scanner = new Scanner(System.in);
+    public static final String LANGUAGECODE = "абвгґдеєжзиіїйклмнопрстуфхцчшщьюя .,\":-!?";
+    public static final String MENU_INFO =
```
use Java text blocks

Volodymyr95 commented on src/Constants.java:
```diff
@@ -0,0 +1,11 @@
+import java.util.Scanner;
+
+public class Constants {
+    public static final Scanner scanner = new Scanner(System.in);
```
don't create it like constant, create a new one where u need it

Volodymyr95 commented on src/Constants.java:
```diff
@@ -0,0 +1,11 @@
+import java.util.Scanner;
+
+public class Constants {
```
make it final and add private constructor

Volodymyr95 commented on src/Encoding.java:
```diff
@@ -5,9 +5,8 @@
 import static java.nio.file.Files.writeString;
 
-public class Encoding {
+public class Encoding extends Constants{
```
don't extends constant

Volodymyr95 commented on src/FirstOperation.java:
```diff
@@ -0,0 +1,32 @@
+
+public class FirstOperation extends Constants{
+
+    public List<String> ChooseTheFile(){
```
methods name should start from lower case letter

Volodymyr95 commented on src/Menu.java:
```diff
@@ -0,0 +1,34 @@
+        List<String> list ;
+
+            //Вибір напрямку програми
+            if (in.equals("1")) {
```
"1","2" and other numbers should be like constants:
private(or public) static final String ENCODING_ITEM = "1";

Commit:
Rebild Main structure of programm:
Big changes.

Commit:
Rebild Main structure of programm:
Big changes:
-Add Keys;
-Add BruteForce(and make him work);
-Add AnalyzorForCrypt;
-Change structure of Menu;
and some little things.

Volodymyr95 commented on src/AnalyzorForCrypt.java:
```diff
@@ -0,0 +1,29 @@
+
+            }
+        }
+        if (count == 1) {
```
1,2, are magic numbers

Volodymyr95 commented on src/AnalyzorForCrypt.java:
```diff
@@ -0,0 +1,29 @@
+        for (int i = 1; i < 4; i++) {
+            if (Objects.equals(keyCharStar, copyOfList[i])){
+                count++;
+            } else if (Objects.equals(keyCharPersent, copyOfList[i])) {
```
not needed equals for char

Volodymyr95 commented on src/BruteForce.java:
```diff
@@ -0,0 +1,14 @@
+    public BruteForce(String list){
+        int number = new AnalyzorForCrypt().analyzorForCrypt(list);
+        switch(number){
+            case 3 -> new KeyOne(list);
```
3,5,7,9 are magic numbers

Volodymyr95 commented on src/Decoding.java:
```diff
@@ -1,40 +1,23 @@
-                }
+
+        for (char c : copyOfList) {
+            // (char)(c - number);
```
comented code

Volodymyr95 commented on src/Encoding.java:
```diff
@@ -1,44 +1,48 @@
-        char[] result = code.toCharArray();
+    public Encoding(String list, int number){
+        char[] copyOfList = list.toCharArray();
         Path path = Path.of("d://result.txt");
```
hard-coded url, it should be passed from console(user input)

Volodymyr95 commented on src/FirstOperation.java:
```diff
@@ -0,0 +1,34 @@
+        try {
+            if (exists(path)) {
+                List<String> list = readAllLines(path);
+                return list;
```
return readAllLines(path);

Volodymyr95 replied:
not fixed still

Volodymyr95 commented on src/FirstOperation.java:
```diff
@@ -0,0 +1,34 @@
+
+    private Scanner scanner = new Scanner(System.in);
+    public List<String> chooseTheFile(){
+        System.out.println("Введіть шлях до файлу: ");
```
use English

Volodymyr95 commented on src/KeyFour.java:
```diff
@@ -0,0 +1,5 @@
+public class KeyFour {
+    public KeyFour(String list){
+        new Decoding(list, 9);
```
9 is magic number

Volodymyr95 commented on src/Menu.java:
```diff
@@ -0,0 +1,48 @@
+        String menuChooseItem = menuIn.nextLine();
+        List<String> list ;
+
+            //Вибір напрямку програми
```
delete comments

Commit:
Another Rebild, no more "magic number" and other changes.

Volodymyr95 commented on src/AnalyzorForCrypt.java:
```diff
@@ -1,28 +1,32 @@
-public class AnalyzorForCrypt {
 
+public class AnalyzorForCrypt extends Encoding{
+    private int count = 0;
```
make it constants

Volodymyr95 commented on src/AnalyzorForCrypt.java:
```diff
@@ -1,28 +1,32 @@
                 count++;
-            } else if (Objects.equals(keyCharPersent, copyOfList[i])) {
+            }else if (copyOfList[i] == keyCharPersent) {
                 count += 2;
```
2 is magic number

Volodymyr95 commented on src/BruteForce.java:
```diff
@@ -1,14 +1,15 @@
-            case 7 -> new KeyTree(list);
-            case 9 -> new KeyFour(list);
-            default -> System.out.println("I don`t know how to help you!");
+        if(number == stepByThree){ new KeyOne(list);
```
u can refactor it using map

8Invader8 replied:
I was try to remake "if" to "HashMap", but in final rezult, I`ll have worst and bigger code than I had.
Or I don`t understand correct what I should do

Commit:
Final changes

Volodymyr95 commented on src/AnalyzorForCrypt.java:
```diff
@@ -0,0 +1,34 @@
+    private static final char KEY_CHAR_STAR = '*';
+    private static final char KEY_CHAR_PERSENT = '%';
+
+    public int analyzorForCrypt(String list){
```
list, is not informative name for String

Volodymyr95 commented on src/AnalyzorForCrypt.java:
```diff
@@ -0,0 +1,34 @@
+    public int analyzorForCrypt(String list){
+        char[] copyOfList = list.toCharArray();
+
+        for (int i = 1; i < 5; i++) {
```
why 5? 5 is magic number

Volodymyr95 commented on src/Decoding.java:
```diff
@@ -1,40 +1,24 @@
-
-import static java.nio.file.Files.writeString;
 
 public class Decoding extends Encoding{
```
Decoding is not Encoding, inheritance you can use for 'is a' relationship

Volodymyr95 commented on src/Encoding.java:
```diff
@@ -1,44 +1,58 @@
     }
 
     public Encoding() {
+
```
empty constructor should be first

Volodymyr95 commented on src/FirstOperation.java:
```diff
@@ -0,0 +1,34 @@
+import static java.nio.file.Files.exists;
+import static java.nio.file.Files.readAllLines;
+
+public class FirstOperation {
```
FirstOperation is not informative name

Volodymyr95 commented on src/KeyFour.java:
```diff
@@ -0,0 +1,8 @@
+public class KeyFour extends Encoding{
+    private int keyNine = 9;
```
what keyNine mean?

Volodymyr95 commented on src/KeyFour.java:
```diff
@@ -0,0 +1,8 @@
+public class KeyFour extends Encoding{
+    private int keyNine = 9;
+    protected int stepForString = 5;
```
what stepForString mean?

Commit:
(Find some comment`s from mentor that didn`t saw)
Changes:
-Rename mistakes for more understanding:
-Remake program logic inside program;
-move files to packages;
Program -works, but have one mistake that, don`t understand how to fix it:
-When use BruteForce, program ask write path to file for write result More times than it needs. I need a help!

Commit:
Fix Problem!
A more little changes of names and minor fix!

Volodymyr95 commented on src/api/Main.java:
```diff
@@ -0,0 +1,9 @@
+public class Main {
+    public static void main(String[] args) {
+
+        new Menu().Run();
```
why method name start from upper letter?

Volodymyr95 commented on src/api/Menu.java:
```diff
@@ -0,0 +1,10 @@
+
+    public void Run(){
+        while (MenuItem.isRunning){
+            new MenuItem();
```
not needed create here object of menu item, u can move code from menu item here

Volodymyr95 commented on src/api/crypting/AnalyzorForCrypt.java:
```diff
@@ -0,0 +1,36 @@
+            }
+        }
+
+        if (count == KEY_ONE) {
```
use switch case

Volodymyr95 commented on src/api/crypting/FileAndKeyForCrypt.java:
```diff
@@ -0,0 +1,34 @@
+        } catch (Exception e) {
+        System.out.println("Oh, you have to a Big one. So, do something!" + e);
+        }
+        return null;
```
don't return null it's a bad practice

Volodymyr95 commented on src/api/crypting/SecretKey.java:
```diff
@@ -0,0 +1,45 @@
+                throw new RuntimeException(e);
+            }
+
+        } else if (keyForCrypt == secondKeyForStep) {
```
don't use == for comparing

Commit:
Change:
1) '==" to equals(in SecretKey);
2) if to switch(In AnalyzorForCrypt);
3) little change for work;
4)change 'return null' on return emptyList(for it works);
5)remove from MenuItem to Menu(delete MenuItem).
